### DISCUSSION TITLE
Looking for advice on i18n support (both webpack 4 and 5)

### DISCUSSION THREAD

**Discussion by charvey77** [webpack/webpack#13829] [Q&A]
Hello!

We currently have a build system that is using webpack 4. Due to business requirements, we are unable to upgrade to webpack 5 at the moment (This is a side issue). We are looking for advice on the best way to handle i18n support in our build system, with the future in mind as well. We've seen that the [i18n-loader](https://github.com/webpack-contrib/i18n-loader) has been deprecated. The best option we have found in our research seems to be [i18n-webpack-plugin](https://github.com/zainulbr/i18n-webpack-plugin). 

In the webpack documentation (both for webpack 4 and 5), there are brief references to Moment.js, but no real detail on how to set up Moment.js to help with locales either. (And it seems that Moment is no longer being actively maintained too).

Does anyone here have recommendations on other solutions, or can you confirm that i18n-webpack-plugin is our best available option?

Thanks!
-Chris

**Comment by alexander-akait** [comment]
I recommend to use non build solution for i18, do you use react/vue/pure js?

**Reply by charvey77** [reply]
Our developers are pure JS. I'm on an infrastructure team that is responsible for building/bundling the JS code for production. We've been using the dojo-webpack-plugin for handling the i18n side of things, but there is an effort to move away from Dojo localization files in our company, so we have to figure out the best way for webpack bundles to handle i18n files going forward.

**Reply by alexander-akait** [reply]
Make sense to look at https://github.com/i18next/i18next

**Reply by charvey77** [reply]
Excellent, thanks very much!
